pulp_container/app/viewsets.py

```diff
@@ -201,6 +201,71 @@ class ContainerRepositoryViewSet(RepositoryViewSet):
     endpoint_name = "container"
     queryset = models.ContainerRepository.objects.all()
     serializer_class = serializers.ContainerRepositorySerializer
+    permission_classes = (AccessPolicyFromDB,)
+    queryset_filtering_required_permission = "container.view_containerrepository"
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:container.add_containerrepository",
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.view_containerrepository",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.delete_containerrepository",
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.change_containerrepository",
+            },
+            {
+                "action": ["sync"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:container.sync_containerrepository",
+                    "has_remote_param_model_or_obj_perms:container.view_containerremote",
+                ],
+            },
+            {
+                "action": ["add", "remove", "tag", "untag", "copy_tags", "copy_manifests"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:container.modify_content_containerrepository",
```

Comment:
Do we need to define permissions for the added content, too?

Reply:
i don't understand how that would work? in this plugin we do not have the modify action

Reply:
Should it be this way for every action?
```
{
                "action": ["tag"],
                "principal": "authenticated",
                "effect": "allow",
                "condition": "has_model_or_obj_perms:container.tag_containerrepository",
            },
```

Reply:
theoretically all of the actions on the repo sync/tag/untag/add/remove/copy/ qualify you to view its content

Reply:
build_image is another action of the repo we need to register

Reply:
That is exactly the question here: We need to define any non-statandard (crud) permission on the model class. [0]
Do we want a separate permission for each action on the viewset, or do we want to collect them into a single "modify_content" permission? Or should "tag" and "untag" be combined?

[0] https://github.com/pulp/pulp_container/pull/179/files#diff-1707426fbe1933a12da3490e42f0dbceea79e47f7c883708c0b0cb49755b1d94R219-R222

Reply:
@mdellweg I have read some more docs and i think having these actions grouped will suffice.